sparkle/src/components/Avatar.tsx

```diff
@@ -1,93 +1,176 @@
+  for (let i = 0; i < name.length; i++) {
+    hash = name.charCodeAt(i) + ((hash << 5) - hash);
+  }
+  const colors = [
+    "s-bg-red-300",
+    "s-bg-orange-300",
+    "s-bg-amber-300",
+    "s-bg-yellow-300",
+    "s-bg-lime-300",
+    "s-bg-green-300",
+    "s-bg-emerald-300",
+    "s-bg-teal-300",
+    "s-bg-cyan-300",
+    "s-bg-sky-300",
+    "s-bg-blue-300",
+    "s-bg-indigo-300",
+    "s-bg-violet-300",
+    "s-bg-purple-300",
+    "s-bg-fuchsia-300",
+    "s-bg-rose-300",
+  ];
+  return colors[Math.abs(hash) % colors.length];
+};
+
+const getTextVariant = (name: string) => {
+  if (/\+/.test(name)) {
+    return "s-text-slate-700";
+  }
+  let hash = 0;
+  for (let i = 0; i < name.length; i++) {
+    hash = name.charCodeAt(i) + ((hash << 5) - hash);
+  }
+  const txtColors = [
+    "s-text-red-700",
+    "s-text-orange-700",
+    "s-text-amber-700",
+    "s-text-yellow-700",
+    "s-text-lime-700",
+    "s-text-green-700",
+    "s-text-emerald-700",
+    "s-text-teal-700",
+    "s-text-cyan-700",
+    "s-text-sky-700",
+    "s-text-blue-700",
+    "s-text-indigo-700",
+    "s-text-violet-700",
+    "s-text-purple-700",
+    "s-text-fuchsia-700",
+    "s-text-rose-700",
+  ];
+  return txtColors[Math.abs(hash) % txtColors.length];
+};
+
+interface AvatarProps {
+  size?: AvatarSizeType;
   name?: string;
   emoji?: string;
   visual?: string | React.ReactNode;
-  clickable?: boolean;
   onClick?: () => void;
+  clickable?: boolean;
   busy?: boolean;
   isRounded?: boolean;
   backgroundColor?: string;
   className?: string;
   disabled?: boolean;
-};
-
-const colors = [
-  "s-bg-red-300",
-  "s-bg-orange-300",
-  "s-bg-amber-300",
-  "s-bg-yellow-300",
-  "s-bg-lime-300",
-  "s-bg-green-300",
-  "s-bg-emerald-300",
-  "s-bg-teal-300",
-  "s-bg-cyan-300",
-  "s-bg-sky-300",
-  "s-bg-blue-300",
-  "s-bg-indigo-300",
-  "s-bg-violet-300",
-  "s-bg-purple-300",
-  "s-bg-fuchsia-300",
-  "s-bg-rose-300",
-];
-
-const txtColors = [
-  "s-text-red-700",
-  "s-text-orange-700",
-  "s-text-amber-700",
-  "s-text-yellow-700",
-  "s-text-lime-700",
-  "s-text-green-700",
-  "s-text-emerald-700",
-  "s-text-teal-700",
-  "s-text-cyan-700",
-  "s-text-sky-700",
-  "s-text-blue-700",
-  "s-text-indigo-700",
-  "s-text-violet-700",
-  "s-text-purple-700",
-  "s-text-fuchsia-700",
-  "s-text-rose-700",
-];
-
-const sizeClasses = {
-  xs: "s-h-6 s-w-6",
-  sm: "s-h-8 s-w-8",
-  md: "s-h-10 s-w-10",
-  lg: "s-h-16 s-w-16",
-  xl: "s-h-20 s-w-20",
-  xxl: "s-h-36 s-w-36",
-  auto: "s-w-full s-relative",
-};
-
-const roundedClasses = {
-  xs: "s-rounded-lg",
-  sm: "s-rounded-xl",
-  md: "s-rounded-2xl",
-  lg: "s-rounded-3xl",
-  xl: "s-rounded-[28px]",
-  xxl: "s-rounded-[38px]",
-  auto: "s-rounded-[30%]",
-};
-
-const textSize = {
-  xs: "s-text-xs",
-  sm: "s-text-sm",
-  md: "s-text-base",
-  lg: "s-text-3xl",
-  xl: "s-text-5xl",
-  xxl: "s-text-7xl",
-  auto: "s-text-xl",
-};
+}
 
 export function Avatar({
-  size = "md",
+  size,
```

Comment:
Why forcing size and classname?

Reply:
They are actually optional and size gets a default value in the cva